mlir-tensorrt/tensorrt/lib/TensorRT/Transforms/BroadcastElimination.cpp

```diff
@@ -263,6 +263,11 @@ static bool broadcastAbsorbPreconditions(BroadcastOp broadcastOp) {
 }
 
 namespace {
+
+static bool isOneOrDynamic(int64_t dimSize) {
+  return dimSize == 1 || dimSize < 0;
```

Comment:
use `ShapedType::isDynamic(dimSize)`. The specific sentinel value representing dynamic shape is an implementation detail.